src/components/Clusters/ClusterDetail.tsx

```diff
@@ -1,5 +1,140 @@
+import { Button, Card, CardSection, Group, Text } from '@mantine/core';
+import { useState } from 'react';
+import { Helmet } from 'react-helmet-async';
+import { RiDeleteBin5Line, RiSettings4Line, RiToggleFill, RiToggleLine } from 'react-icons/ri';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useAuth } from '@/hooks/useAuth';
+import { useClusterDetails, useJobsOnCluster } from '@/hooks/useClusters';
+import { disableModal, enableModal } from '@/utils/modals';
+import { pluralize } from '@/utils/string';
+
+import ClusterConfigCard from './ClusterConfigCard';
+import ClusterJobsCard from './ClusterJobsCard';
+import ClusterLabelBadge from './ClusterLabelBadge';
+import ClusterOwnerDetails from './ClusterOwnerDetails';
+
 function ClusterDetail() {
-  return <div className="flex h-full flex-col" />;
+  const navigate = useNavigate();
+  const { clusterId: clusterName = '' } = useParams();
+  if (!clusterName) navigate('/404', { replace: true });
+
+  const { data: cluster = null } = useClusterDetails(clusterName);
+  const [enabled, setEnabled] = useState(!cluster?.disabled);
```

Comment:
After adding that 404 check. From that point on, we know cluster is non-null so no need to have ?. in the rest of this component.